pkg/core/native/native_neo.go

```diff
@@ -447,33 +439,21 @@ func (n *NEO) PostPersist(ic *interop.Context) error {
 	return nil
 }
 
-func (n *NEO) getGASPerVote(d *dao.Simple, key []byte, indexes []uint32) []big.Int {
-	sort.Slice(indexes, func(i, j int) bool {
-		return indexes[i] < indexes[j]
-	})
-	start := make([]byte, 4)
-	binary.BigEndian.PutUint32(start, indexes[len(indexes)-1])
-
-	need := len(indexes)
-	var reward = make([]big.Int, need)
-	collected := 0
-	d.Seek(n.ID, storage.SeekRange{
-		Prefix:    key,
-		Start:     start,
-		Backwards: true,
-	}, func(k, v []byte) bool {
-		if len(k) == 4 {
-			num := binary.BigEndian.Uint32(k)
-			for i, ind := range indexes {
-				if reward[i].Sign() == 0 && num <= ind {
-					reward[i] = *bigint.FromBytes(v)
-					collected++
-				}
-			}
-		}
-		return collected < need
-	})
-	return reward
+func (n *NEO) getLatestGASPerVote(d *dao.Simple, key []byte) big.Int {
+	var g big.Int
+	cacheKey := string(key[1:])
+	cache := d.GetROCache(n.ID).(*NeoCache)
+	if g, ok := cache.gasPerVoteCache[cacheKey]; ok {
+		return g
+	}
+	item := d.GetStorageItem(n.ID, key)
+	if item == nil {
+		g = *big.NewInt(0)
+	} else {
+		g = *bigint.FromBytes(item)
+	}
+	cache.gasPerVoteCache[cacheKey] = g
```

Comment:
Be careful with cache, we need RW here, it's important (currently it's RO).

Reply:
Will do

Reply:
Done

Reply:
I'd do this in a bit different way. Always take RO cache (it's much more efficient this way) and fallback to getting from DAO here without updating the cache. The cache will be updated in a normal way when the time comes in `PostPersist` and on a proper running node this entry will always be there, so it'll be accessible via RO cache and that's it.

Reply:
> and fallback to getting from DAO here without updating the cache.

Then we won't use cache inside the `getLatestGASPerVote` at all, so it's always DAO's value? It's the only place where the cached value is retrieved, why do we need the cache for this value then?

Reply:
> Then we won't use cache inside the getLatestGASPerVote at all

Why? It'll be used because effectively it'll always be there, updated on every committee update in `PostPersist`.

Reply:
I'm confusing a bit about it. So the only place where cached value should be updated is inside the `PostPersist`?

Reply:
Yeah and that's enough because that's where this value is really changed, no other code can ever touch it (except for reading of course).